### PR TITLE
Add IF EXISTS and IF NOT EXISTS checks to ALTER TABLE

### PR DESCRIPTION
**Pull request by leetcode-1533**
Fixes https://github.com/prestosql/presto/issues/2260

### CODE REVIEW

File: presto-parser/src/main/antlr4/io/prestosql/sql/parser/SqlBase.g4 [hunk 1]
```diff
@@ -55,15 +55,15 @@ statement
...
-        DROP COLUMN column=qualifiedName                               #dropColumn
-    | ALTER TABLE tableName=qualifiedName
-        ADD COLUMN column=columnDefinition                             #addColumn
+    | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
+        RENAME COLUMN (IF EXISTS)? from=identifier TO to=identifier                 #renameColumn
+    | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
+        DROP COLUMN (IF EXISTS)? column=qualifiedName                               #dropColumn
+    | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
+        ADD COLUMN (IF NOT EXISTS)? column=columnDefinition                             #addColumn
     | ANALYZE qualifiedName (WITH properties)?                         #analyze
```
Comment (kokosing) [inline presto-parser/src/main/antlr4/io/prestosql/sql/parser/SqlBase.g4:67]:
```suggestion
    | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
        RENAME COLUMN (IF EXISTS)? from=identifier TO to=identifier    #renameColumn
    | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
        DROP COLUMN (IF EXISTS)? column=qualifiedName                  #dropColumn
    | ALTER TABLE (IF EXISTS)? tableName=qualifiedName
        ADD COLUMN (IF NOT EXISTS)? column=columnDefinition            #addColumn
    | ANALYZE qualifiedName (WITH properties)?                         #analyze
```

File: presto-parser/src/main/java/io/prestosql/sql/tree/AddColumn.java [hunk 2]
```diff
@@ -27,22 +27,26 @@
 {
     private final QualifiedName name;
     private final ColumnDefinition column;
+    private final boolean tableNameExists;
```
Comment (kokosing) [inline presto-parser/src/main/java/io/prestosql/sql/tree/AddColumn.java:30]: tableExists here and in other statement classes

File: presto-parser/src/main/java/io/prestosql/sql/tree/AddColumn.java [hunk 3]
```diff
@@ -27,22 +27,26 @@
...
+    private final boolean columnNameNotExists;
```
Comment (kokosing) [inline presto-parser/src/main/java/io/prestosql/sql/tree/AddColumn.java:31]: `columnNotExists` here and in other statement classes

In other words `Name` is not misleading and should not be used in these variable names

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java [hunk 4]
```diff
@@ -375,6 +379,7 @@ public void testRenameColumn()
         assertUpdate("CREATE TABLE " + tableName + " AS SELECT 'some value' x", 1);
 
         assertUpdate("ALTER TABLE " + tableName + " RENAME COLUMN x TO y");
+        assertUpdate("ALTER TABLE " + tableName + " RENAME COLUMN IF EXISTS columnNotExists TO y");
```
Comment (kokosing) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:382]:
```suggestion
        assertUpdate("ALTER TABLE " + tableName + " RENAME COLUMN x TO before_y");
        assertUpdate("ALTER TABLE " + tableName + " RENAME COLUMN IF EXISTS before_y TO y");
        assertUpdate("ALTER TABLE " + tableName + " RENAME COLUMN IF EXISTS columnNotExists TO y");
```

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java [hunk 5]
```diff
@@ -395,9 +404,17 @@ public void testDropColumn()
         assertUpdate("CREATE TABLE " + tableName + " AS SELECT 123 x, 111 a", 1);
 
         assertUpdate("ALTER TABLE " + tableName + " DROP COLUMN x");
+        assertUpdate("ALTER TABLE " + tableName + " DROP COLUMN IF EXISTS notExistColumn");
```
Comment (kokosing) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:407]: Add test where actually column is dropped when `IF EXISTS` is used.

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java [hunk 6]
```diff
@@ -407,6 +424,7 @@ public void testAddColumn()
         assertUpdate("CREATE TABLE " + tableName + " AS SELECT CAST('first' AS varchar) x", 1);
 
         assertQueryFails("ALTER TABLE " + tableName + " ADD COLUMN x bigint", ".* Column 'x' already exists");
+        assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN IF NOT EXISTS x bigint");
```
Comment (kokosing) [inline presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:427]: call this `assertUpdate(...)` twice

File: presto-parser/src/main/java/io/prestosql/sql/tree/AddColumn.java (hunk 3 above)
Comment (leetcode-1533) [reply presto-parser/src/main/java/io/prestosql/sql/tree/AddColumn.java:31]: fixed, thanks!

File: presto-parser/src/main/java/io/prestosql/sql/tree/AddColumn.java (hunk 2 above)
Comment (leetcode-1533) [reply presto-parser/src/main/java/io/prestosql/sql/tree/AddColumn.java:30]: fixed thanks!

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java (hunk 5 above)
Comment (leetcode-1533) [reply presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:408]: fixed thanks!

Comment (leetcode-1533) [reply presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:408]: updated for other test case as well.

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java (hunk 6 above)
Comment (leetcode-1533) [reply presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:428]: updated. For knowledge purpose, can you tell me why should we call "assertUpdate("ALTER TABLE " + tableName + " ADD COLUMN IF NOT EXISTS x bigint");" twice?

**Comment by leetcode-1533** [comment]
HI, all comments have been addressed. If needed, I can also squash and rebase the commits

File: presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java (hunk 6 above)
Comment (kokosing) [reply presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java:438]: Oh my bad. My point was to run it twice so once it creates a column and then it skips column creation (suppress the error0. But here column `x` already exists. So to make it work here we need to use different column name than `x`.

Also it would be better to use something else than `x` something that would express the test case. Maybe `x` -> `testing_add_column_if_not_exists`?

Comment (leetcode-1533) [reply presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java]: updated! Already add a statement to test can add column even with "if exists" statement, so made a duplicate of which.

**Comment by leetcode-1533** [comment]
Hi, all comments addressed, feel free to let me know if there is any issue!

**Comment by kokosing** [comment]
Merged, thanks!
